docs/source-pytorch/advanced/training_tricks.rst

```diff
@@ -191,29 +191,52 @@ The algorithm in short works by:
 Customizing Batch Size Finder
 =============================
 
-You can also customize the :class:`~pytorch_lightning.callbacks.batch_size_finder.BatchSizeFinder` callback to run at different epochs. This feature is useful while fine-tuning models since
-you can't always use the same batch size after unfreezing the backbone.
+1. You can also customize the :class:`~pytorch_lightning.callbacks.batch_size_finder.BatchSizeFinder` callback to run
+   at different epochs. This feature is useful while fine-tuning models since you can't always use the same batch size after
+   unfreezing the backbone.
 
 .. code-block:: python
 
-        from pytorch_lightning.callbacks import BatchSizeFinder
+    from pytorch_lightning.callbacks import BatchSizeFinder
 
 
-        class FineTuneBatchSizeFinder(BatchSizeFinder):
-            def __init__(self, *args, **kwargs):
-                super().__init__(*args, **kwargs)
-                self.milestones = milestones
+    class FineTuneBatchSizeFinder(BatchSizeFinder):
+        def __init__(self, milestones, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.milestones = milestones
 
-            def on_fit_start(self, *args, **kwargs):
-                return
+        def on_fit_start(self, *args, **kwargs):
+            return
 
-            def on_train_epoch_start(self, trainer, pl_module):
-                if trainer.current_epoch in self.milestones or trainer.current_epoch == 0:
-                    self.scale_batch_size(trainer, pl_module)
+        def on_train_epoch_start(self, trainer, pl_module):
+            if trainer.current_epoch in self.milestones or trainer.current_epoch == 0:
+                self.scale_batch_size(trainer, pl_module)
 
 
-        trainer = Trainer(callbacks=[FineTuneBatchSizeFinder(milestones=(5, 10))])
-        trainer.fit(...)
+    trainer = Trainer(callbacks=[FineTuneBatchSizeFinder(milestones=(5, 10))])
+    trainer.fit(...)
+
+
+2. Run batch size finder for ``validate``/``test``/``predict``.
+
+.. code-block:: python
+
+    from pytorch_lightning.callbacks import BatchSizeFinder
+
+
+    class EvalBatchSizeFinder(BatchSizeFinder):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+        def on_fit_start(self, *args, **kwargs):
+            return
+
+        def on_test_start(self, trainer, pl_module):
+            self.scale_batch_size(trainer, pl_module)
+
+
+    trainer = Trainer(callbacks=[EvalBatchSizeFinder()])
```

Comment:
I thought this wasn't necessary and instead you would pass `method="test"`

Reply:
that is for `trainer.tune(... , method='test')`

Reply:
can be a little confusing.. let's discuss the `tune` API future after release.

Reply:
My question is more along the lines of "When would you want to subclass vs using `tune(..., method="test")`"?

Are you saying we would be removing `.tune()` which would make this snippet the recommended approach?

Reply:
for validate/test/predict, this would be same.. i.e using `trainer.tune(method='test')` would do the job.. 
but now since we have 2 ways to do the same thing, I was thinking if we could remove `trainer.tune` altogether since callbacks can do the job now. The reason I added callback in the example instead of tune is so that in case we move forward with removing it in the future, users should start using callbacks from now since we are not going to remove these callbacks.

Reply:
if we keep `trainer.tune, I'll update the docs adding this API approach.

Reply:
Thanks for the clarification! Removing tune makes sense to me but that should be discussed in a separate issue.

Reply:
yeap!